app/__init__.py

```diff
@@ -34,106 +40,197 @@
+                               mimetype='application/json')
+        except Exception as e:
+            get_connection().db_update(identity, response.WORKSPACE_FAILED)
+            app.logger.error(e)
+            return jsonify(status=errors.ERROR_STATUS, message=str(e))
+    except (KeyError, ValueError, TypeError):
+        return jsonify(status=errors.ERROR_STATUS,
+                       message=errors.KEY_ERROR_NAME)
+
 
 # Route for listing all workspaces
 @app.route('/workspace/list', methods=['GET'])
-def linchpin_list_workspace():
+def linchpin_list_workspace() -> Response:
+    """
+        GET request route for listing workspaces.
+        :return : response with a list of workspaces
+        from the destination set in config.py
+    """
     try:
-        workspace_array = []
+        workspace_array = get_connection().db_list_all()
         # path specifying location of working directory inside server
-        for x in os.listdir(os.path.join(app.root_path + WORKING_DIR)):
-            if os.path.isdir(x):
-                workspace_dict = {'name ': x}
-                workspace_array.append(workspace_dict)
         return Response(json.dumps(workspace_array), status=200,
                         mimetype='application/json')
     except Exception as e:
         app.logger.error(e)
-        return jsonify(status=409, message=str(e))
+        return jsonify(status=errors.ERROR_STATUS, message=str(e))
 
 
-@app.route('/workspace/delete', methods=['POST'])
-def linchpin_delete_workspace():
+@app.route('/workspace/list/<name>', methods=['GET'])
+def linchpin_list_workspace_by_name(name) -> Response:
+    """
+        GET request route for listing workspaces.
+        :return : response with a list of workspaces
+        from the destination set in config.py
+    """
     try:
-        data = request.json  # Get request body
-        name = data["name"]
+        workspace = get_connection().db_search(name)
         # path specifying location of working directory inside server
-        for x in os.listdir(os.path.join(app.root_path + WORKING_DIR)):
-            if x == name:
-                shutil.rmtree(name)
-                return jsonify(name=name,
-                               status="Workspace deleted successfully")
-        return jsonify(status="Workspace " + name + " not found")
+        return Response(json.dumps(workspace), status=200,
+                        mimetype='application/json')
     except Exception as e:
         app.logger.error(e)
-        return jsonify(status=409, message=str(e))
+        return jsonify(status=errors.ERROR_STATUS, message=str(e))
 
 
-@app.route('/workspace/fetch', methods=['POST'])
-def linchpin_fetch_workspace():
+@app.route('/workspace/delete/<identity>', methods=['DELETE'])
+def linchpin_delete_workspace(identity) -> Response:
+    """
+        DELETE request route for deleting workspaces.
+        RequestBody: {"name": "workspacename"}
```

Comment:
@mansikulkarni96 it would be little confusing to have both identity and workspace name separately. 
If we have id , we do not need workspace name. Since workspace name can be searched with id from database

Reply:
yes I don't think requestbody is needed for this now, I will remove it.